pallets/identity/src/lib.rs

```diff
@@ -342,17 +342,34 @@ decl_module! {
             target_account: T::AccountId,
             secondary_keys: Vec<secondary_key::api::SecondaryKey<T::AccountId>>
         ) -> DispatchResult {
-            // Sender has to be part of CDDProviders
             let cdd_id = Self::ensure_origin_call_permissions(origin)?.primary_did;
+            Self::base_cdd_register_did(cdd_id, target_account, secondary_keys)?;
+            Ok(())
```

Comment:
```suggestion
            Self::base_cdd_register_did(cdd_id, target_account, secondary_keys)
```

Reply:
The return types are different () vs IdentityId.